### PR TITLE
Feature CLS2 433 mark task as complete

### PR DESCRIPTION
**Pull request by marijnkampf**
## Description of change

You are now able to mark an Investment Task as complete.

## Test instructions

Create an [investment project task](http://localhost:3000/tasks/bb135bd6-d135-4d9a-8b65-5b478eeabe73) for a [companies' investment project](http://localhost:3000/investments/projects/ca5aed0b-70fa-4a57-8a5c-6a11bebdd644/tasks).
The page were you can view the Investment project task now contains a 'Mark as complete' button. This archives the task.

### Before

image

### After

image

Clicking the button should show the Flash message: 

image

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'

- [x] Has the branch been rebased to main?
- [x] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

File: src/client/modules/Tasks/TaskDetails/tasks.js [hunk 1]
```diff
@@ -1,4 +1,18 @@
+import { METHOD_POST } from '../../../../common/constants'
 import { apiProxyAxios } from '../../../components/Task/utils'
 
 export const getTaskDetail = (id) =>
   apiProxyAxios.get(`v4/task/${id}`).then(({ data }) => data)
+
+export const archiveTask = ({ taskId }) => {
```
Comment (chopkinsmade) [inline src/client/modules/Tasks/TaskDetails/tasks.js:7]: Can one line this like the above task 
```
export const archiveTask = ({ taskId }) =>
  apiProxyAxios.post(`/v4/task/${taskId}/archive`, { reason: 'completed' })
```

File: src/client/modules/Tasks/TaskDetails/TaskButtons.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          id="archive-task-form"
+          analyticsFormName="archiveTaskForm"
+          redirectTo={() =>
+            urls.investments.projects.tasks.index(
+              task.investmentProjectTask.investmentProject.id
+            )
+          }
+          cancelRedirectTo={false}
+          submissionTaskName={TASK_ARCHIVE_TASK}
+          transformPayload={(values) => ({
```
Comment (chopkinsmade) [inline src/client/modules/Tasks/TaskDetails/TaskButtons.jsx:23]: We don't have FormField components in the rendering of this form, so we can just get rid of the values:
```
transformPayload={() => ({
            taskId: task.id,
          })}
```

File: src/client/modules/Tasks/TaskDetails/index.jsx [hunk 3]
```diff
@@ -27,8 +27,8 @@ const TaskDetails = ({ task }) => {
         {
           text: taskTitle,
         },
-        company?.id,
-        company?.name
+        company?.id ? company.id : '',
```
Comment (chopkinsmade) [inline src/client/modules/Tasks/TaskDetails/index.jsx:30]: Found this pairing with caitlin earlier, can simplify the ternary to this
```
company?.id || '',
company?.name || ''
```

File: src/apps/routers.js [hunk 4]
```diff
@@ -69,6 +69,7 @@ const reactRoutes = [
   '/omis/:orderId/edit/internal-details',
   '/investments/projects/:projectId/tasks/create',
   '/investments/projects/:projectId/tasks',
+  '/tasks/:taskId/archive',
```
Comment (chopkinsmade) [inline src/apps/routers.js:72]: We don't have a page for the archiving it's just an endpoint, so no need for a react route

File: src/lib/urls.js [hunk 5]
```diff
@@ -651,5 +651,6 @@ module.exports = {
   },
   tasks: {
     details: url('/tasks', '/:taskId'),
+    archive: url('/tasks', '/:taskId/archive'),
```
Comment (chopkinsmade) [inline src/lib/urls.js:654]: Same as the routers.js file comment, I don't think we need this

File: test/component/cypress/specs/client/modules/tasks/TaskDetails/TaskButtons.cy.jsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+
+import { taskWithInvestmentProjectFaker } from '../../../../../../../functional/cypress/fakers/task'
+import TaskDetailsTable from '../../../../../../../../src/client/modules/Tasks/TaskDetails/TaskDetailsTable'
+import DataHubProvider from '../../../../provider'
+
+describe('Task buttons', () => {
+  const Component = (props) => (
+    <DataHubProvider>
+      <TaskDetailsTable {...props} />
```
Comment (chopkinsmade) [inline test/component/cypress/specs/client/modules/tasks/TaskDetails/TaskButtons.cy.jsx:10]: This test file is for the TaskButtons but it's mounting a TaskDetailsTable. This test should mount the TaskButtons directly without needing to mount the wrapping component

File: src/client/modules/Tasks/TaskDetails/index.jsx (hunk 3 above)
Comment (marijnkampf) [reply src/client/modules/Tasks/TaskDetails/index.jsx:30]: That's a nice solution.

**Review by dredmonds — APPROVED** [review]

**Pull request merged**